solutions/devicesimulation-nohub/single-vm/setup.sh

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env bash
+# Copyright (c) Microsoft. All rights reserved.
+# Note: Windows Bash doesn't support shebang extra params
+set -ex
+
+APP_PATH="/app"
+WEBUICONFIG="${APP_PATH}/webui-config.js"
+WEBUICONFIG_SAFE="${APP_PATH}/webui-config.js.safe"
+WEBUICONFIG_UNSAFE="${APP_PATH}/webui-config.js.unsafe"
+ENVVARS="${APP_PATH}/env-vars"
+DOCKERCOMPOSE="${APP_PATH}/docker-compose.yml"
+CERTS="${APP_PATH}/certs"
+CERT="${CERTS}/tls.crt"
+PKEY="${CERTS}/tls.key"
+
+REPOSITORY="https://raw.githubusercontent.com/Azure/pcs-cli/azure-iot-pcs-simulation/solutions/devicesimulation-nohub/single-vm"
+SCRIPTS_URL="${REPOSITORY}/scripts/"
+
+# ========================================================================
+
+export HOST_NAME="${1:-localhost}"
+export PCS_STORAGEADAPTER_DOCUMENTDB_CONNSTRING="${3}"
+export PCS_CERTIFICATE="${4}"
+export PCS_CERTIFICATE_KEY="${5}"
+export PCS_AUTH_AUDIENCE="${6}"
+export PCS_AUTH_ISSUER="https://sts.windows.net/${7}/"
+export PCS_WEBUI_AUTH_TYPE="aad"
+export PCS_WEBUI_AUTH_AAD_APPID="${6}"
+export PCS_WEBUI_AUTH_AAD_TENANT="${7}"
+export PCS_WEBUI_AUTH_AAD_INSTANCE="${8}"
+
+# ========================================================================
+
+# Configure Docker registry based on host name
+# ToDo: we may need to add similar parameter to AzureGermanCloud and AzureUSGovernment
+config_for_azure_china() {
+    set +e
+    local host_name=$1
+    if (echo $host_name | grep -c  "\.cn$") ; then
+        # If the host name has .cn suffix, dockerhub in China will be used to avoid slow network traffic failure.
+        local config_file='/etc/docker/daemon.json'
+        echo "{\"registry-mirrors\": [\"https://registry.docker-cn.com\"]}" > ${config_file}
+        service docker restart
+
+        # Rewrite the AAD issuer in Azure China environment
+        export PCS_AUTH_ISSUER="https://sts.chinacloudapi.cn/$2/"
+    fi
+    set -e
+}
+
+config_for_azure_china $HOST_NAME $5
```

Comment:
In the script for RM 2.0, the 5th parameter is `AAD tenant ID`. We passed $5 to function `config_for_azure_china` to build the issuer for MC. While in this script, `AAD tenant ID` is the 7th parameter. So this line should be changed.

Reply:
good catch, I'll fix that